src/ypd/model/project.py

```diff
@@ -19,43 +28,48 @@ class Project(Base, HasUserMixin):
     archived = Column(Boolean)
     needsReview = Column(Boolean)
 
+    @with_session
+    def post(self, title, description, poster, session=None):
+        """Posts this project to the database
 
-    def post(self, title, description, poster):
-        """Posts this project to the database"""
+        Args:
+            title (str): Project title
+            description (str): Project description
+            poster (User): User who posted the project
+        
+        Kwargs:
+            session (Session): session to perform the query on. Supplied by decorator
+        """
         self.title = title
     
         self.description = description
         self.poster_id = poster.id
         self.date = datetime.utcnow() #we might want to only assign this when the project is approved
         self.archived = False
         self.needsReview = False #TODO: When we implement the review workflow, we'll set this to True here
-
-        session = Session()
         session.add(self)
-        session.commit()
-        session.close()
         
-    @classmethod    
-    def get(cls, id):
+    @classmethod  
+    @with_session  
+    def get(cls, id, session=None):
         """Gets projects from database by id
         
         Args:
             id (int): id of the project to get
+
+        Kwargs:
+            session (Session): session to perform the query on. Supplied by decorator
             
         Returns:
             The project with the given id
             
         Raises:
             ValueError: If no project with the given id exists
         """
-        session = Session()
-        # Look for project ID in database
         try:
-            result = session.query(cls).filter_by(id = id).one()
+            return session.query(cls).filter_by(id = id).one()
         except NoResultFound as e:
             raise ValueError(f'No project found with id {id}') from e
```

Comment:
is that f supposed to be in the argument? (f'No project found with id {id}')

Reply:
Yeah, it's an f-string. It's equivalent to a str.format to pass variables into a string like "Hello, my name is {}".format(name)

Reply:
Alright, I just wanted clarification before approving the merge to make sure it was intentional.

Reply:
Yea, LeQuient's right. I, personally, favor f-strings because they're easy to read